terraform/mgmtcluster.tf

```diff
@@ -54,14 +65,18 @@ resource "openstack_compute_instance_v2" "mgmtcluster_server" {
 
   network { port = openstack_networking_port_v2.mgmtcluster_port.id }
   block_device {
-    uuid                  = data.openstack_images_image_ids_v2.images.ids[0]
-    source_type           = "image"
-    volume_size           = 30
+    uuid                  = openstack_blockstorage_volume_v3.mgmtcluster_volume.id
+    source_type           = "volume"
```

Comment:
I assume there is no way to name the volume if we use source_type = "image" here?

Reply:
The issue is not the source_type but for some reason inside the instance you can't name the volume. It can be only done when the instance is defined as separate resource.